### PR TITLE
Setup project using classes

### PR DESCRIPTION
**Pull request by yonas44**
Hello👋, for this PR;
- I created `Person` class with  `private` and `public` methods.
- I created `Student` class from `Person` which also adds a custom attribute and method.
- I created `Teacher` class from `Person` but modifies the method `can_use_services?`.

Thank you for reviewing my PR!

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class Person
+  attr_accessor :name, :age, :parent_permission
+
+  def initialize(name, age, parent_permission)
```
Comment (prolajumokeoni) [inline main.rb:5]: - [x] Kindly define your id as  an `attr_reader :id` and set `@id = Random.rand(1..1000)` in your constructor, this is part of the project requirement

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative './main'
+
+class Teacher < Person
+  def initialize(specialization)
+    super()
```
Comment (prolajumokeoni) [inline teacher.rb]:
```suggestion
 def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
```

- [x] Kindly add `age, name, parent_permission` as parameters in your constructor, this is part of the project requirement

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @yonas44 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Git flow was correctly used 
✔️  Good PR title and Description



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by yonas44** [comment]
Okay thank you for all your feedback🙏

**Review by prolajumokeoni — APPROVED** [review]
Hi @yonas44

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀  
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**